core/metric/violation.py

```diff
@@ -1,10 +1,37 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, List
+
+
+@dataclass
+class Location:
+    file: Path
+    first_line: int
+    last_line: int
```

Comment:
Looks good.
General proposition. We discussed introducing token count as well, that is how far into the first line does the violation actually begin and likewise for the end of the violation. This doesn't hinder us from merging but is just an idea. This is nice since we can simply add it without changing much. I like it.

👍